src/utils/filter_helpers.py

```
@@ -0,0 +1,158 @@
+import pandas as pd
+from typing import List
+
+
+def create_notnull_mask(df: pd.DataFrame, col: str) -> pd.Series:
+    """Return a mask for string values in column col that are not null."""
+    return df[col].str.len() > 0
+
+
+def check_cols_in_df(df: pd.DataFrame, cols: List[str]) -> None:
+    """CHeck that a col exists in a dataframe, if not raise an error.
+
+    Args:
+        df (pd.DataFRame): The dataframe
+        cols List[str]: The names of the columns to be checked
+
+    Raises:
+
+    """
+    msg = ""
+    for col in cols:
+        if col not in df.columns:
+            msg += f"Column {col} does not exist in this dataframe. "
+    if msg != "":
+        raise ValueError(msg)
+
+
+def get_clear_status_mask(df: pd.DataFrame) -> pd.Series:
+    check_cols_in_df(df, ["status"])
```

General comment: if the column exists, this returns an error.
Didn't we said that you wanted also to have the possibility of applying the filter but with the option of avoiding the error of the column does not exist? in this case, whouldn't it make sense to add another boolean argument called, say, "error_if_no_column", which, if a column doesn't exist, makes the "check_cols_in_df" skip the filter rather than throwing an error?

Or did you want to avoid an error if even a column on which you were not filtering, didn't exist? In this case, I think you are implementing it correctly.

---

it's the latter, and the way the other functions are set up will allow us to do everything we want to.
I think we really do want to throw an error if the code tells us to filter on a column and it doesn't exist.